File: civitasJAVA/src/civitas/CasillaCalle.java

package civitas;
import java.util.ArrayList;


public class CasillaCalle extends Casilla {
    private TituloPropiedad titulo;
    
    //Constructor
    CasillaCalle(TituloPropiedad _titulo)
    {
        super(_titulo.getNombre());
        titulo = _titulo;
    }
    
    //CasillaCalle recibe a jugador
    @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(jugadorCorrecto(actual, todos))
        {
            informe(actual, todos);
            Jugador jugador = todos.get(actual);

            if (!titulo.tienePropietario()) {
                if(jugador.puedeComprarCasilla())
                    System.out.println("La casilla no está comprada: jugador puede comprar casilla");
            }
            else
                titulo.tramitarAlquiler(jugador);
        }
    }
    
    //toString
    @Override
    public String toString()
    {
        return super.toString()+ "\n\tTitulo propiedad = " + (titulo == null ? "null" : titulo.toString());
    }
    
    TituloPropiedad getTituloPropiedad() { return titulo; }
}

File: civitasRUBY/lib/casilla_calle.rb
#encoding: utf-8

require_relative "casilla"

module Civitas

  class CasillaCalle < Casilla   
    def initialize(titulo)
      super(titulo.nombre)
      @tituloPropiedad = titulo
    end
    
    #RecibeJugador de calle
    def recibeJugador(actual, todos)
      if jugadorCorrecto(actual, todos)
        informe(actual, todos)
        jugador=todos.at(actual)
        
        if !@tituloPropiedad.tienePropietario
          if jugador.puedeComprarCasilla
            puts "La casilla no está comprada: jugador puede comprar casilla"
          end
        else
          @tituloPropiedad.tramitarAlquiler(jugador)
        end
      end
    end
    
    #toString
    def toString
      return "{\n\tCasilla:  #{@nombre}" +
             (@tituloPropiedad!=nil ? "\n\tNumHoteles:  #{@tituloPropiedad.numHoteles}" : "") +
             (@tituloPropiedad!=nil ? "\n\tNumCasas:  #{@tituloPropiedad.numCasas}" : "")     +
             "\n}"
    end
    
  end

end